src/game_interpreter_map.cpp

```diff
@@ -212,57 +225,54 @@ bool Game_Interpreter_Map::ContinuationEnemyEncounter(RPG::EventCommand const& c
 
 	continuation = NULL;
 
-	switch (Game_Temp::battle_result) {
-	case Game_Temp::BattleVictory:
-		if ((Game_Temp::battle_defeat_mode == 0 && Game_Temp::battle_escape_mode != 2) || !SkipTo(Cmd::VictoryHandler, Cmd::EndBattle)) {
-			index++;
-			return false;
-		}
-		index++;
-		return true;
-	case Game_Temp::BattleEscape:
-		switch (Game_Temp::battle_escape_mode) {
-		case 0:	// disallowed - shouldn't happen
-			return true;
-		case 1:
+	int sub_idx = subcommand_sentinel;
+
+	if (Game_Temp::battle_result == Game_Temp::BattleVictory) {
+		sub_idx = 0;
+	}
+
+	if (Game_Temp::battle_result == Game_Temp::BattleEscape) {
+		sub_idx = 1;
+		//FIXME: subidx set before this anyway??
+		if (Game_Temp::battle_escape_mode == 1) {
 			return CommandEndEventProcessing(com);
-		case 2:
-			if (!SkipTo(Cmd::EscapeHandler, Cmd::EndBattle)) {
-				index++;
-				return false;
-			}
-			index++;
-			return true;
-		default:
-			return false;
 		}
-	case Game_Temp::BattleDefeat:
-		switch (Game_Temp::battle_defeat_mode) {
-		case 0:
+	}
+
+	if (Game_Temp::battle_result == Game_Temp::BattleDefeat) {
+		sub_idx = 2;
+		//FIXME: subidx set before this anyway??
+		if (Game_Temp::battle_defeat_mode == 0) {
 			return CommandGameOver(com);
-		case 1:
-			if (!SkipTo(Cmd::DefeatHandler, Cmd::EndBattle)) {
-				index++;
-				return false;
-			}
-			index++;
-			return true;
-		default:
-			return false;
-		}
-	case Game_Temp::BattleAbort:
-		if (!SkipTo(Cmd::EndBattle)) {
-			index++;
-			return false;
 		}
-		index++;
-		return true;
-	default:
-		return false;
 	}
+
+	SetSubcommandIndex(com.indent, sub_idx);
+
+	return true;
+}
+
+bool Game_Interpreter_Map::CommandVictoryHandler(RPG::EventCommand const& com) { // code 20710
+	return CommandOptionGeneric(com, 0, {Cmd::EscapeHandler, Cmd::DefeatHandler, Cmd::EndBattle});
+}
+
+bool Game_Interpreter_Map::CommandEscapeHandler(RPG::EventCommand const& com) { // code 20711
+	return CommandOptionGeneric(com, 1, {Cmd::DefeatHandler, Cmd::EndBattle});
+}
+
+bool Game_Interpreter_Map::CommandDefeatHandler(RPG::EventCommand const& com) { // code 20712
+	return CommandOptionGeneric(com, 2, {Cmd::EndBattle});
```

Comment:
Where do these numbers that are the 2nd parameter for CommandOptionGeneric come from?

Reply:
These are the values that `subcommand_path` has to be to trigger this action. 

There are example frame dumps in #1706

This deserves an enum to help make the code more self documenting.